### PR TITLE
Make the job ID copying easier

### PR DESCRIPTION
**Pull request by pdostal**
A lot of online systems allow user to either copy to clipboard or easily select the job/ticket/case ID.
My intention is to enable the same functionality for openQA, so when user wants to f.e. clone the job it will be faster.

There is [fa-copy](https://fontawesome.com/icons/copy?f=classic&s=solid) icon in All tests and in Job group overview and there is job ID printed in job overview and is clickable.

image
image
image

### CODE REVIEW

File: assets/javascripts/tests.js [hunk 1]
```diff
@@ -554,3 +556,9 @@ function showJobDependency(deps) {
   }
   return result;
 }
+
+$(document).on('click', '.copy-jobid', function (event) {
+  event.preventDefault();
+  var id = $(this).attr('data-jobid');
+  navigator.clipboard.writeText(id);
```
Comment (Martchus) [inline assets/javascripts/tests.js:563]: Use of `var` should be avoided in new code, as well as use of jQuery when it is useless anyways.

```suggestion
  navigator.clipboard.writeText(this.dataset.jobid);
```

File: assets/javascripts/tests.js [hunk 2]
```diff
@@ -57,6 +57,8 @@ function renderTestName(data, type, row) {
 
   var html = '';
   if (is_operator) {
+    html += ' <a class="copy-jobid" href data-jobid="' + row.id + '">';
```
Comment (Martchus) [inline assets/javascripts/tests.js:60]:
```suggestion
    html += '<a class="copy-jobid" href data-jobid="' + row.id + '">';
```

File: templates/webapi/test/infopanel.html.ep [hunk 3]
```diff
@@ -4,6 +4,7 @@
                 % if (current_route 'latest') {
                     <%= link_to $job->id => url_for ('test', testid => $job->id) %>:
                 % }
+                <i>#<a class="copy-jobid" href data-jobid="<%= $job->id %>"><%= $job->id %></a></i>
```
Comment (asdil12) [inline templates/webapi/test/infopanel.html.ep:7]:
```suggestion
                <i><a class="copy-jobid" href data-jobid="<%= $job->id %>">#<%= $job->id %></a></i>
```

Looks better and makes more sense if hash is part of the id.
Also you could set the href to link to the current job.

**Review by asdil12 — CHANGES_REQUESTED** [review]

**Comment by pdostal** [comment]
Resolved. I also added `padding-right: 5px;` so the icons are a bit further from the stop / restart icon. 
image

**Comment by asmorodskyi** [comment]
LGTM!

**Review by perlpunk — APPROVED** [review]

File: assets/javascripts/tests.js [hunk 4]
```diff
@@ -57,6 +57,8 @@ function renderTestName(data, type, row) {
 
   var html = '';
   if (is_operator) {
+    html += '<a class="copy-jobid" href="' + urlWithBase('/tests/' + row.id) + '" data-jobidi="' + row.id + '">';
```
Comment (b10n1k) [inline assets/javascripts/tests.js:60]: i believe *jobidi* has a typo! dir you mean `jobid`?

**Review by b10n1k — CHANGES_REQUESTED** [review]

File: assets/javascripts/tests.js [hunk 5]
```diff
@@ -57,6 +57,8 @@ function renderTestName(data, type, row) {
...
+    html += '<i class="action fa fa-fw fa-copy" title="Copy job id"></i></a>';
```
Comment (okurz) [inline assets/javascripts/tests.js:61]: This is the full "job URL", not just ID, right?

**Review by okurz — CHANGES_REQUESTED** [review]

File: assets/javascripts/tests.js (hunk 5 above)
Comment (pdostal) [reply assets/javascripts/tests.js:61]: Yes, the `<a href=` points to the full job id but if you're javascript is enabled then it will not be followed.
That was suggested by @asdil12 [here](https://github.com/os-autoinst/openQA/pull/5249#discussion_r1263890961).

**Review by b10n1k — APPROVED** [review]

File: assets/javascripts/tests.js (hunk 5 above)
Comment (okurz) [reply assets/javascripts/tests.js:61]: I am not sure I understand. But if the end result is the full job URL then please call it as such in code as well as in the help and git commit message

Comment (pdostal) [reply assets/javascripts/tests.js:61]: In the `href=""` is the full test URL in case your javascript is off.
But if the javascript is working then only the job id is put to your clipboard.

Comment (pdostal) [reply assets/javascripts/tests.js:61]: If you need job URL it's two clicks to copy it. But if you f.e. want to clone 10 different jobs, then it is very time consuming to copy the job URL, parse the job ID, store the job ID and repeat 10 times. So this button only gets you the job id.

Comment (pdostal) [reply assets/javascripts/tests.js:61]: Also people usually reuse commands from their history, so for example `openqa-clone-job https://openqa.opensuse.org/tests/11560397 --host https://pdostal-server.suse.cz --show-progress --skip-chained-deps` and if I want to do 5 verification runs it's usually only the job ID to replace.

**Review by grisu48 — APPROVED** [review]

File: assets/javascripts/tests.js (hunk 5 above)
Comment (okurz) [reply assets/javascripts/tests.js:61]: Yeah, sorry, I don't think that's a use case I would like feature *more* prominently to just copy the ID instead of the URL as that sounds like a rather personal choice of how you process command lines. Instead what I suggest to do commonly is to iterate over full job URLs, something like:

```bash
for i in https://openqa.opensuse.org/tests/11560397 https://openqa.opensuse.org/tests/11560234 https://openqa.opensuse.org/tests/11560402; do
openqa-clone-job $i --host https://pdostal-server.suse.cz --show-progress --skip-chained-deps
done
```

Comment (pdostal) [reply assets/javascripts/tests.js:61]: Your command work when you want to clone bunch of jobs at once but if I want to clone only one job and next job in half an hour you end up replacing the URL over and over.

Then let's say I want to monitor a job in my command line: `openqa-cli api --osd --pretty jobs/$job_id`
Oh and then I have review and I want to use the awesome [grisu48/openqa-mon](https://github.com/grisu48/openqa-mon): `openqa-mon http://openqa.opensuse.org -j $ID1,$ID2,$ID3,...`

I've so far received quite positive feedback.
If the icon is in wrong place, we can move it.

Basically every system dealing with jobs or tickets displays the ID somewhere. And I think there's a reason for that. See:
Screenshot 2023-07-18 at 11 48 28
Screenshot 2023-07-18 at 11 46 40
Screenshot 2023-07-18 at 11 44 58
Screenshot 2023-07-18 at 11 44 03
Screenshot 2023-07-18 at 12 01 32

Comment (grisu48) [reply assets/javascripts/tests.js:61]: I also think this feature might turn out to be useful to more folk than just Pavel :slightly_smiling_face:

**Comment by pdostal** [comment]
I just got rid of the `<a href="JOBID"` in favour of `<a href="#"` so the link is still valid but the code looks less confusing.

**Review by mimi1vx — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

File: assets/javascripts/tests.js (hunk 5 above)
Comment (okurz) [reply assets/javascripts/tests.js:61]: Ok, convinced :D

**Review by asdil12 — APPROVED** [review]

**Review by Martchus — APPROVED** [review]

**Comment by Martchus** [comment]
Removed the review request for @b10n1k as the mentioned typo has been fixed and there are more than enough reviews at this point.
